app/src/main/java/org/p2p/wallet/bridge/claim/interactor/ClaimInteractor.kt

```diff
@@ -1,11 +1,26 @@
 package org.p2p.wallet.bridge.claim.interactor
 
+import org.p2p.core.token.SolAddress
 import org.p2p.ethereumkit.external.repository.EthereumRepository
 import org.p2p.wallet.bridge.claim.repository.EthereumClaimRepository
+import org.p2p.wallet.bridge.model.BridgeBundle
 import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
 
 class ClaimInteractor(
     private val ethereumClaimRepository: EthereumClaimRepository,
     private val ethereumRepository: EthereumRepository,
     private val tokenKeyProvider: TokenKeyProvider
-)
+) {
+    suspend fun getEthereumBundle(erc20TokenAddress: String, amount: String): BridgeBundle {
```

Comment:
i think erc20Token address is hex?

Reply:
amount -> amountInLamports

Reply:
hex is String =)

Reply:
@gslevinkov erc20Token will be in EthAddress wrap class

Reply:
tokenToClaim.publicKey, 

tokenToClaim == Token.Eth

Token.Eth publicKey == String

data class Eth constructor(
        override val publicKey: String,
        val totalInUsd: BigDecimal?,
        val total: BigDecimal,
        override val tokenSymbol: String,
        override val decimals: Int,
        override val mintAddress: String,
        override val tokenName: String,
        override val iconUrl: String?,
        override var rate: BigDecimal?,
        override var currency: String = Constants.USD_READABLE_SYMBOL
    )
    
    you want to make it EthAddress ? or wrap it in presenter from string?